doc/examples/segmentation/plot_trainable_segmentation.py

```diff
@@ -0,0 +1,168 @@
+are then labeled from the prediction of the classifier.
+
+This segmentation algorithm is called trainable segmentation in other software
+such as ilastik [2]_ or ImageJ [3]_ (where it is also called "weka
+segmentation").
+
+.. [1] https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html
+.. [2] https://www.ilastik.org/documentation/pixelclassification/pixelclassification
+.. [3] https://imagej.net/Trainable_Weka_Segmentation#Training_features_.282D.29
+"""
+
+
+from itertools import combinations_with_replacement
+import itertools
+import numpy as np
+import matplotlib.pyplot as plt
+from skimage import io, filters, feature, segmentation
+from skimage import img_as_float32
+from sklearn.ensemble import RandomForestClassifier
+
+
+def _features_sigma(img, sigma, intensity=True, edges=True, texture=True):
+    """Features for a single value of the Gaussian blurring parameter ``sigma``
+    """
+    features = []
+    img_blur = filters.gaussian(img, sigma)
+    if intensity:
+        features.append(img_blur)
+    if edges:
+        features.append(filters.sobel(img_blur))
+    if texture:
+        H_elems = [np.gradient(np.gradient(img_blur)[ax0], axis=ax1)
+            for ax0, ax1 in combinations_with_replacement(range(img.ndim), 2)]
+        eigvals = feature.hessian_matrix_eigvals(H_elems)
+        for eigval_mat in eigvals:
+            features.append(eigval_mat)
+    return features
+
+
+def _compute_features_gray(img, intensity=True, edges=True, texture=True,
+                          sigma_min=0.5, sigma_max=16):
+    """Features for a single channel image. ``img`` can be 2d or 3d.
+    """
+    # computations are faster as float32
+    img = img_as_float32(img)
+    sigmas = np.logspace(np.log2(sigma_min), np.log2(sigma_max),
+            num=int(np.log2(sigma_max) - np.log2(sigma_min) + 1), base=2, endpoint=True)
+    n_sigmas = len(sigmas)
+    all_results = [_features_sigma(img, sigma, intensity=intensity, edges=edges, texture=texture) for sigma in sigmas]
+    return list(itertools.chain.from_iterable(all_results))
+
+
+def compute_features(img, multichannel=True,
+                          intensity=True, edges=True, texture=True,
+                          sigma_min=0.5, sigma_max=16):
+    """Features for a single- or multi-channel image.
+    """
+    if img.ndim == 3 and multichannel:
+        all_results = (_compute_features_gray(
+                img[..., dim], intensity=intensity, edges=edges, texture=texture,
+                sigma_min=sigma_min, sigma_max=sigma_max) for dim in range(img.shape[-1]))
+        features = list(itertools.chain.from_iterable(all_results))
+    else:
+        features = _compute_features_gray(
+            img[..., dim], intensity=intensity, edges=edges, texture=texture,
+            sigma_min=sigma_min, sigma_max=sigma_max)
+    return np.array(features)
+
+
+def trainable_segmentation(img, mask, clf, multichannel=True,
```

Comment:
Can we rename `mask` to `labels`? mask is always a boolean array in the library.